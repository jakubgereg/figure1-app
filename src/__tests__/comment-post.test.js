import React from "react";
import ReactDOM from "react-dom";
import CommentPost from "../components/comment-post";
import { shallow } from "enzyme";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CommentPost />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders date text correctly", () => {
  const wrapper = shallow(
    <CommentPost
      commentDate={new Date(2018, 1, 1, 12, 30, 40)}
      username="Jakub"
    />
  );
  const should = <div> on 2/1/2018 said:</div>;
  expect(wrapper.contains(should)).toEqual(true);
});

it("renders a snapshot", () => {
  const wrapper = shallow(
    <CommentPost
      commentDate={new Date(2018, 1, 1, 12, 30, 40)}
      username="Jakub"
    />
  );
  expect(wrapper).toMatchSnapshot();
});
