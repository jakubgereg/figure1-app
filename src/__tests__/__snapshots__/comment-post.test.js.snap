// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentPost
    commentDate={2018-02-01T17:30:40.000Z}
    username="Jakub"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="author-header"
        >
          <Author
            username="Jakub"
          />
          <div>
             on 
            2/1/2018
             said:
          </div>
        </div>,
        <div
          className="comment-content"
        />,
      ],
      "className": "comment-post",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Author
              username="Jakub"
            />,
            <div>
               on 
              2/1/2018
               said:
            </div>,
          ],
          "className": "author-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "image": undefined,
              "username": "Jakub",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " on ",
                "2/1/2018",
                " said:",
              ],
            },
            "ref": null,
            "rendered": Array [
              " on ",
              "2/1/2018",
              " said:",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "comment-content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="author-header"
          >
            <Author
              username="Jakub"
            />
            <div>
               on 
              2/1/2018
               said:
            </div>
          </div>,
          <div
            className="comment-content"
          />,
        ],
        "className": "comment-post",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Author
                username="Jakub"
              />,
              <div>
                 on 
                2/1/2018
                 said:
              </div>,
            ],
            "className": "author-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "image": undefined,
                "username": "Jakub",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " on ",
                  "2/1/2018",
                  " said:",
                ],
              },
              "ref": null,
              "rendered": Array [
                " on ",
                "2/1/2018",
                " said:",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "comment-content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
